client/src/components/rooms/RoomsMeetingExperience.tsx

```diff
@@ -2,33 +2,228 @@
 import { Theme, PartialTheme, Stack, Spinner } from '@fluentui/react';
 import { fullSizeStyles } from '../../styles/Common.styles';
 import { RoomParticipantRole, RoomsInfo } from '../../models/RoomModel';
-import { PostCallConfig } from '../../models/ConfigModel';
+import { PostCallConfig, TranscriptionClientOptions } from '../../models/ConfigModel';
 import { Survey } from '../postcall/Survey';
 import MobileDetect from 'mobile-detect';
 import InviteInstructions from './InviteInstructions';
 import { isRoomsInviteInstructionsEnabled, isRoomsPostCallEnabled } from '../../utils/MeetingExperienceUtil';
+import {
+  connectToCallAutomation,
+  fetchTranscriptionStatus,
+  getCallSummaryFromServer,
+  LocaleCode,
+  startTranscription,
+  stopTranscription,
+  SummarizeResult
+} from '../../utils/CallAutomationUtils';
+import { Call, TeamsCall } from '@azure/communication-calling';
+import { SlideTextEdit20Regular } from '@fluentui/react-icons';
+import { TranscriptionOptionsModal } from './transcriptionNotifications/TranscriptionOptionsModal';
+import { CustomNotifications } from './transcriptionNotifications/CustomNotifications';
 
 export interface RoomsMeetingExperienceProps {
   roomsInfo: RoomsInfo;
   token: string;
   postCall?: PostCallConfig;
   fluentTheme?: PartialTheme | Theme;
   onDisplayError(error: any): void;
+  // this needs to be re-thought. we need a way for the config to say we are not using transcription, but also
+  // auto start, and transcription only
+  transcriptionClientOptions?: TranscriptionClientOptions;
 }
 
 const RoomsMeetingExperience = (props: RoomsMeetingExperienceProps): JSX.Element => {
-  const { roomsInfo, token, postCall, fluentTheme, onDisplayError } = props;
+  const { roomsInfo, token, postCall, fluentTheme, onDisplayError, transcriptionClientOptions } = props;
   const { userId, userRole, locator } = roomsInfo;
+  const [transcriptionStarted, setTranscriptionStarted] = useState(false);
+  const [showTranscriptionModal, setShowTranscriptionModal] = useState(false);
+  const [summarizationLanguage, setSummarizationLanguage] = useState<LocaleCode>('en-US');
+  const [summary, setSummary] = useState<SummarizeResult>();
+  const [summarizationStatus, setSummarizationStatus] = useState<'None' | 'InProgress' | 'Complete'>('None');
+  const [statefulClient, setStatefulClient] = useState<StatefulCallClient>();
+  const [callAgent, setCallAgent] = useState<DeclarativeCallAgent>();
+  const [call, setCall] = useState<Call | TeamsCall | undefined>();
+  const [callAdapter, setCallAdapter] = useState<CommonCallAdapter>();
+  const [customNotications, setCustomNotifications] = useState<ActiveNotification[]>([]);
+  const [callConnected, setCallConnected] = useState(false);
+  const [serverCallId, setServerCallId] = useState<string | undefined>(undefined);
+  const [transcriptionStartedByYou, setTranscriptionStartedByYou] = useState(false);
```

Comment:
Wow that's a lot of states! hahah. 
Do we want to have a state manager, or at the bare minimum have these alphabetized ?

Reply:
I reorganized it alphabetically, I want to keep this simple with using react useStates rather than adding another manager to keep track of these things.